vscode/src/telemetry.ts

```diff
@@ -250,3 +250,19 @@ export function sendTelemetryEvent<E extends keyof EventTypes>(
     )}`,
   );
 }
+
+function getBrowserRelease(): string {
+  if (navigator.userAgentData) {
+    const browser =
+      navigator.userAgentData.brands[navigator.userAgentData.brands.length - 1];
+    return `${browser.brand}/${browser.version}`;
+  } else {
+    return "";
+  }
+}
+
+export function getUserAgent(context: vscode.ExtensionContext): string {
```

Comment:
There's a LOT of plumbing through the context everywhere in this PR. Being that you already need to pass the context to `initTelemetry` (line 227 above) which already constructs a singleton (`reporter`) for use in any calls to send telemetry (line 237 above), could you not just construct the user agent string once in `initTelemetry` (it's not going to change for the lifetype of the app) and use it generally in `sendTelemetryEvent`?

Or do we absolutely specifically only want this propery on Azure network request events?